youtube_dl/extractor/common.py

```diff
@@ -2194,7 +2194,11 @@ def geo_verification_headers(self):
         return headers
 
     def _generic_id(self, url):
-        return compat_urllib_parse_unquote(os.path.splitext(url.rstrip('/').split('/')[-1])[0])
+        id = compat_urllib_parse_unquote(os.path.splitext(url.rstrip('/').split('/')[-1])[0])
+        # If our id matches a stopword, use the 2nd-to-last URL path component instead
+        if id in ['story']:
+            id = compat_urllib_parse_unquote(os.path.splitext(url.rstrip('/').split('/')[-2])[0])
```

Comment:
Reverted, but I don't understand why...It doesn't seem correct to:
. Create a custom extractor for bostonglobe.com, given that it just another BrightCove site.
. Extract videos all with the id 'story' when there is a simple fix
I'm not sure what qualifies as an 'ad hoc hack' versus a good general fix here. I would have categorized this one differently than you did. Is there something to be done? Happy to put it in a separate PR, also.